17.05 Edge docs

mstanleyjones opened the pull request:
### Proposed changes

Include Edge content for 17.05 release

### Unreleased project version (optional)

Docker CE 17.05

Fixes #2755 
Fixes #2999

mstanleyjones commented:
@alexellis PTAL

alexellis commented on engine/userguide/eng-image/multistage-build.md:
```diff
@@ -0,0 +1,145 @@
+read and maintain.
+
+> **Acknowledgment**:
+> Special thanks to Alex Ellis for granting permission to use his blog post
```
@mstanleyjones can you include a link out for the name?

+> Special thanks to [Alex Ellis](https://twitter.com/alexellisuk) for granting permission to use his blog post

alexellis commented on engine/userguide/eng-image/multistage-build.md:
```diff
@@ -0,0 +1,145 @@
+contained everything needed to build your application), and a slimmed-down one
+to use for production, which only contained your application and exactly what
+was needed to run it. This has been referred to as the "builder
+pattern".Maintaining two Dockerfiles is not ideal.
```
Should there be a space here between `pattern".M` ?

+pattern".Maintaining two Dockerfiles is not ideal.

mstanleyjones commented:
Addressed your feedback, @alexellis 

I also noticed in your example you don't clean up the extracted `./app`  from your local filesystem (which is presumably a Git repo where you wouldn't want to commit an artifact) and you don't prune the intermediate image. Not sure if that matters to you.

alexellis commented on engine/userguide/eng-image/multistage-build.md:
```diff
@@ -0,0 +1,158 @@
+
+echo Building alexellis2/href-counter:latest
+
+docker build --no-cache -t alexellis2/href-counter:latest .  
```
@mstanleyjones let's add an `rm ./app` here so that things are belt and braces. I think I just maintain a .gitignore.

alexellis commented:
For pruning - I don't think that's necessary for the intermediate image in the builder pattern, after all I think is also going to turn into a legacy way to build images and wouldn't normally be pushed anywhere.
